### ISSUE TITLE
render_callback, disappearing array index if default value of an attribute is set in JS via registerBlockType

### ISSUE THREAD

**Issue by andreyc0d3r** [WordPress/gutenberg#10641]
**Describe the bug**
Not sure if this is a bug. But it quacks like a bug, so....
So I have a block with a 'render_callback'

```
register_block_type(
            $this->namespace . '/' . $this->slug,
            array(
                'editor_script'     => $this->slug . '-editor',
                'editor_style'      => $this->slug . '-editor',
                'render_callback'   => array( $this, 'render_feed_block' )
            )
        );
```

here is the render function

```
function render_feed_block( $attributes ) {
        return print_r( $attributes );
 }
```

in my JS file

```
wp.blocks.registerBlockType( window.feed_settings.namespace + '/' + window.feed_settings.slug, {
    title: 'Data Feed',
    category: 'common',
    attributes: {
        isLoading: {
            default: false
        },
        type: {
            type: 'string'
        },
        content: {
            type: 'string'
        }
    },
    edit: WB_Feed_Block,
    save: function() { return null; }
});
```

In the gutenberg Inspector controls I have a SelectControl component

```
                        el( SelectControl, {
                            options: options,
                            value: this.state.type,
                            onChange: ( value ) => {
                                this.setState( { type: value } );
                                this.props.setAttributes( { type: value } );
                            }
                        } )
```

### Expected behavior

Let's focus on the render_feed_block function in PHP. The setup above works and we see the index 'type' from return print_r( $attributes ); 

The problem occurs when I set a default value to the type attribute within the registerBlockType function in JS like so:

```
        type: {
            type: 'string',
            default: 'news'
        },
```

The addition causes disappearance of the index 'type' from attributes array within this function

```
    function render_feed_block( $attributes ) {

        return print_r( $attributes );

    }
```

index of 'content' is still there, but 'type' is no longer showing up.

Is this expected behavior? shouldn't we always show all of the attributes defined within the JS?

**Comment by ShaMSofT** [comment]
Same here. I expected to have all the attributes in render_callback, but attributes with default value are omitted from the $attributes parameter.

**Comment by dnusca** [comment]
Can confirm this issue on 4.5.1, render_callback will not pull in default attributes. Only once an attribute has been updated with setAttributes will it be available in the $attributes for render_callback.

**Comment by youknowriad** [comment]
Thanks for raising this issue. Yes, default attributes are not persisted which means they're not available in the backend. This is related to the server side awareness of blocks and the work being done in the Block Registration RFC #13693 as this would allow us to get the default values without having them being persisted.

I'm closing this as a duplicate of #7342 for now.
